src/main/java/com/iexec/sms/tee/config/TeeWorkerInternalConfiguration.java

```diff
@@ -21,77 +21,71 @@
 import com.iexec.sms.api.config.SconeServicesProperties;
 import com.iexec.sms.api.config.TeeAppProperties;
 import com.iexec.sms.tee.ConditionalOnTeeFramework;
+import jakarta.validation.Valid;
 import jakarta.validation.constraints.NotBlank;
-import jakarta.validation.constraints.Positive;
+import jakarta.validation.constraints.NotNull;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import org.springframework.util.unit.DataSize;
 import org.springframework.validation.annotation.Validated;
 
+@Slf4j
 @Configuration
 @Validated
 public class TeeWorkerInternalConfiguration {
     @Bean
-    TeeAppProperties preComputeProperties(
-            @Value("${tee.worker.pre-compute.image}")
-            @NotBlank(message = "pre-compute image must be provided")
-            String preComputeImage,
-            @Value("${tee.worker.pre-compute.fingerprint}")
-            @NotBlank(message = "pre-compute fingerprint must be provided")
-            String preComputeFingerprint,
-            @Value("${tee.worker.pre-compute.entrypoint}")
-            @NotBlank(message = "pre-compute entrypoint must be provided")
-            String preComputeEntrypoint,
-            @Value("${tee.worker.pre-compute.heap-size-gb}")
-            @Positive(message = "pre-compute heap size must be provided")
-            long preComputeHeapSizeInGB) {
+    TeeAppProperties preComputeProperties(@Valid @NotNull final TeeWorkerPipelineConfiguration pipelineConfig) {
```

Comment:
The `@Valid` annotation may be a bit overkill here.
With `@Validated` in `TeeWorkerPipelineConfiguration`, we know the application will not start if the configuration is not correct, because the bean will not be created.
This means, this method can already be called if and only if `pipelineConfig` is valid.